pkg/metadata/provider_packet.go

```diff
@@ -84,6 +88,28 @@ func (p *ProviderPacket) Extract() ([]byte, error) {
 
 // networkConfig handles Packet network configuration, primarily bonding
 func networkConfig(ni metadata.NetworkInfo) error {
+	// set up bonding
+	bond, err := netlink.LinkByName("bond0")
+	if err != nil {
+		if strings.Contains(err.Error(), "Link not found") {
+			newBond := new(netlink.Bond)
+			newBond.LinkAttrs.Name = "bond0"
+			newBond.Mode = netlink.BondMode(ni.BondingMode())
+
+			if err := netlink.LinkAdd(newBond); err != nil {
+				if !strings.Contains(err.Error(), "file exists") {
```

Comment:
you should be able to check if the error is EEXIST without using strings.Contains.

I still dont understand why netlink gives this error...